docs/framework/guides/deep-dive/localization.md

```diff
@@ -0,0 +1,240 @@
+	This can be done before initiating the CKEditor 5 editor instance but it requires importing the CKEditor 5 utility function.
+- By [extending the global `window.CKEDITOR_TRANSLATIONS` object](#using-the-windowckeditor_translations-object).
+	This can be done before initiating the CKEditor 5 editor instance.
+- By [creating `.po` files with translations](#creating-po-files) in the `lang/translations/` directory of the published package like other CKEditor 5 packages do.
+	This option will be useful for third-party plugin creators as it allows bundling translations only for needed languages during the webpack compilation.
+
+### Using the `add()` function
+
+The first option for adding translations is via the {@link module:utils/translation-service.add the translation-service's `add()` helper}. This utility adds translations to the `window.CKEDITOR_TRANSLATIONS` object by extending it. Since it needs to be imported, it works only before building the editor.
+
+Starting with the CKEditor 5 v19.0.0 release, it now accepts the `getPluralForm()` function as the third argument. It also accepts an array of translations for a *message* if the *message* should support singular and plural forms.
+
+```js
+add( 'pl', {
+	'Add space': [ 'Dodaj spację', 'Dodaj %0 spacje', 'Dodaj %0 spacji' ]
+} );
+
+// Assuming that the Polish language was picked:
+t( { string: 'Add space', plural: 'Add %0 spaces' }, 1 ) // "Dodaj spację"
+t( { string: 'Add space', plural: 'Add %0 spaces' }, 2 ) // "Dodaj 2 spacje"
+t( { string: 'Add space', plural: 'Add %0 spaces' }, 5 ) // "Dodaj 5 spacji"
+```
+
+### Using the `window.CKEDITOR_TRANSLATIONS` object
+
+The second option is adding translations via the global `window.CKEDITOR_TRANSLATIONS` object.
+
+For each language that should be supported, the `dictionary` property of this object should be extended and the `getPluralForm` function should be provided if missing.
+
+The `dictionary` property is a `message ID ⇒ translations` map, where the `translations` can be either one sentence (a string) or an array of translations with plural forms for the given language if the message should support plural forms as well.
+
+The `getPluralForm` property should be a function that returns the plural form index for a given quantity. Note that when using CKEditor 5 translations, this property will be defined by *CKEditor 5 translation assets*.
+
+Check an example below that demonstrates a part of the `window.CKEDITOR_TRANSLATIONS` object with Polish translations for the `Cancel` and `Add space` *message IDs*:
+
+```js
+{
+	// Each key should be a valid language code.
+	pl: {
+		// A map of translations for the 'pl' language.
+		dictionary: {
+			'Cancel': 'Anuluj',
+			'Add space': [ 'Dodaj spację', 'Dodaj %0 spacje', 'Dodaj %0 spacji' ]
+		},
+
+		// A function that returns the plural form index for the given language.
+		// getPluralForm: n => n == 1 ? 0 : n % 10 >= 2 && n % 10 <= 4 && ( n % 100 < 10 || n % 100 >= 20 ) ? 1 : 2
```

Comment:
I'd uncomment this line, but I'd add another line of comment above reminding that you need to pass this function only if you are adding translations for a new language.